### ISSUE TITLE
Lkm module build

### ISSUE THREAD

**Issue by AldairSoraka** [darlinghq/darling#1061]
**Build Log**
What is the build error?
```
Built target migcom
Built target migexe
Generating osfmk/mach/upl_user.c, osfmk/mach/upl.h, osfmk/mach/upl_server.c, osfmk/mach/upl_server.h, osfmk/mach/uplXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/host_notify_reply_user.c, osfmk/mach/host_notify_reply.h, osfmk/mach/host_notify_reply_server.c, osfmk/mach/host_notify_reply_server.h, osfmk/mach/host_notify_replyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/vm32_map_user.c, osfmk/mach/vm32_map.h, osfmk/mach/vm32_map_server.c, osfmk/mach/vm32_map_server.h, osfmk/mach/vm32_mapXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_notify_user.c, osfmk/mach/mach_notify.h, osfmk/mach/mach_notify_server.c, osfmk/mach/mach_notify_server.h, osfmk/mach/mach_notifyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_voucher_attr_control_user.c, osfmk/mach/mach_voucher_attr_control.h, osfmk/mach/mach_voucher_attr_control_server.c, osfmk/mach/mach_voucher_attr_control_server.h, osfmk/mach/mach_voucher_attr_controlXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/memory_entry_user.c, osfmk/mach/memory_entry.h, osfmk/mach/memory_entry_server.c, osfmk/mach/memory_entry_server.h, osfmk/mach/memory_entryXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/memory_object_control_user.c, osfmk/mach/memory_object_control.h, osfmk/mach/memory_object_control_server.c, osfmk/mach/memory_object_control_server.h, osfmk/mach/memory_object_controlXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/memory_object_user.c, osfmk/mach/memory_object.h, osfmk/mach/memory_object_server.c, osfmk/mach/memory_object_server.h, osfmk/mach/memory_objectXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/memory_object_default_user.c, osfmk/mach/memory_object_default.h, osfmk/mach/memory_object_default_server.c, osfmk/mach/memory_object_default_server.h, osfmk/mach/memory_object_defaultXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_voucher_user.c, osfmk/mach/mach_voucher.h, osfmk/mach/mach_voucher_server.c, osfmk/mach/mach_voucher_server.h, osfmk/mach/mach_voucherXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/vm_map_user.c, osfmk/mach/vm_map.h, osfmk/mach/vm_map_server.c, osfmk/mach/vm_map_server.h, osfmk/mach/vm_mapXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/resource_notify_user.c, osfmk/mach/resource_notify.h, osfmk/mach/resource_notify_server.c, osfmk/mach/resource_notify_server.h, osfmk/mach/resource_notifyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/restartable_user.c, osfmk/mach/restartable.h, osfmk/mach/restartable_server.c, osfmk/mach/restartable_server.h, osfmk/mach/restartableXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/exc_user.c, osfmk/mach/exc.h, osfmk/mach/exc_server.c, osfmk/mach/exc_server.h, osfmk/mach/excXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_exc_user.c, osfmk/mach/mach_exc.h, osfmk/mach/mach_exc_server.c, osfmk/mach/mach_exc_server.h, osfmk/mach/mach_excXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/notify_user.c, osfmk/mach/notify.h, osfmk/mach/notify_server.c, osfmk/mach/notify_server.h, osfmk/mach/notifyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/task_user.c, osfmk/mach/task.h, osfmk/mach/task_server.c, osfmk/mach/task_server.h, osfmk/mach/taskXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/clock_user.c, osfmk/mach/clock.h, osfmk/mach/clock_server.c, osfmk/mach/clock_server.h, osfmk/mach/clockXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/clock_priv_user.c, osfmk/mach/clock_priv.h, osfmk/mach/clock_priv_server.c, osfmk/mach/clock_priv_server.h, osfmk/mach/clock_privXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/processor_user.c, osfmk/mach/processor.h, osfmk/mach/processor_server.c, osfmk/mach/processor_server.h, osfmk/mach/processorXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/host_priv_user.c, osfmk/mach/host_priv.h, osfmk/mach/host_priv_server.c, osfmk/mach/host_priv_server.h, osfmk/mach/host_privXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/host_security_user.c, osfmk/mach/host_security.h, osfmk/mach/host_security_server.c, osfmk/mach/host_security_server.h, osfmk/mach/host_securityXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/lock_set_user.c, osfmk/mach/lock_set.h, osfmk/mach/lock_set_server.c, osfmk/mach/lock_set_server.h, osfmk/mach/lock_setXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_port_user.c, osfmk/mach/mach_port.h, osfmk/mach/mach_port_server.c, osfmk/mach/mach_port_server.h, osfmk/mach/mach_portXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_vm_user.c, osfmk/mach/mach_vm.h, osfmk/mach/mach_vm_server.c, osfmk/mach/mach_vm_server.h, osfmk/mach/mach_vmXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/mach_host_user.c, osfmk/mach/mach_host.h, osfmk/mach/mach_host_server.c, osfmk/mach/mach_host_server.h, osfmk/mach/mach_hostXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/processor_set_user.c, osfmk/mach/processor_set.h, osfmk/mach/processor_set_server.c, osfmk/mach/processor_set_server.h, osfmk/mach/processor_setXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/thread_act_user.c, osfmk/mach/thread_act.h, osfmk/mach/thread_act_server.c, osfmk/mach/thread_act_server.h, osfmk/mach/thread_actXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/mach/clock_reply_user.c, osfmk/mach/clock_reply.h, osfmk/mach/clock_reply_server.c, osfmk/mach/clock_reply_server.h, osfmk/mach/clock_replyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/device/device_user.c, osfmk/device/device.h, osfmk/device/device_server.c, osfmk/device/device_server.h, osfmk/device/deviceXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Generating osfmk/UserNotification/UNDReplyUser.c, osfmk/UserNotification/UNDReply.h, osfmk/UserNotification/UNDReplyServer.c, osfmk/UserNotification/UNDReplyServer.h, osfmk/UserNotification/UNDReplyXtraceMig.c
clang-12: warning: argument unused during compilation: '-arch i386' [-Wunused-command-line-argument]
Built target lkm_generate
Built target rtsig
Built target rtsig_h
Building the Linux kernel module

make[4]: it enter on '/home/aldair/darling/src/external/lkm'
Running kernel version is 5.15.1-3-cacule
rm -f darling-mach.mod.o
make -C /lib/modules/5.15.1-3-cacule/build M=/home/aldair/darling/src/external/lkm modules
make[5]: it enter on '/usr/lib/modules/5.15.1-3-cacule/build'
make[5]: *** There is not any rule to build 'modules'.  Stop.
make[5]: it get outs from '/usr/lib/modules/5.15.1-3-cacule/build'
make[4]: *** [Makefile:615: default] Error 2
make[4]: it get outs from '/home/aldair/darling/src/external/lkm'
make[3]: *** [src/external/lkm/CMakeFiles/lkm.dir/build.make:71: src/external/lkm/CMakeFiles/lkm] Error 2
make[2]: *** [CMakeFiles/Makefile2:58226: src/external/lkm/CMakeFiles/lkm.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:58233: src/external/lkm/CMakeFiles/lkm.dir/rule] Error 2
make: *** [Makefile:20137: lkm] Error 2
```

**System Information**
Artix linux 64bits (Arch linux without systemd)

| Software | Version |
| --- | --- |
| Clang | 12.0.1 |
| CMake | 3.21.4 |
| Linux Kernel | 5.15.1-3 |
| Darling | Git Commit Hash 

i don't know what mean "Git commit hash" but maybe can be 61d1007
also i translate the logs into English with the Google translator

**Comment by Greasy-Monkey** [comment]
i was able to build and install successfully on Arch linux.
```
❯ clang --version
clang version 13.0.0
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
```
❯ uname -a
Linux lenovo 5.15.4-arch1-1 #1 SMP PREEMPT Sun, 21 Nov 2021 21:34:33 +0000 x86_64 GNU/Linux
```
I have BTRFS as root and zfs as home.Faced a little different error {/lib/modules/5.15.3-arch1-1/build: No such file or directory} ; fixed my updating and replacing zfs-dkms with zfs-dkms-git.I don't know if you are using some different file system.Hope it helps !!

**Comment by TylerJaacks** [comment]
BUMP Having the same issue on Ubuntu 21.04 using Clang 13

**Comment by facekapow** [comment]
Closing as this has been resolved and TylerJaacks has opened #1091
